common/evmcompat/helpers.go

```diff
@@ -156,15 +156,19 @@ func recoverAddressFromEos(hash []byte, sig []byte) (common.Address, error) {
 
 func recoverAddressFromEosScatter(hash []byte, sig []byte) (common.Address, error) {
 	var signer common.Address
-	if len(sig) != 67 {
-		return signer, fmt.Errorf("eos signature must be 67 bytes, not %d bytes", len(sig))
+
+	if len(sig) != 107 {
+		return signer, fmt.Errorf("eos scatter signature must be 107 bytes, not %d bytes", len(sig))
 	}
-	signature := ecc.NewSigNil()
-	_, err := signature.Unpack(sig[1:])
+	signature, err := ecc.NewSignature(string(sig[6:]))
 	if err != nil {
 		return signer, fmt.Errorf("cannot unpack eos signature %v", string(sig))
 	}
-	pubKey, err := signature.PublicKey(hash)
+
+	nonceHash := sha256.Sum256([]byte(hex.EncodeToString(sig[:6])))
```

Comment:
sig[:6] is already the hash of the once so should this not be 
nonceHash = sig[:6]

Reply:
yeh it's necessary, `sig` isn't necessarily a hash is just 6 bytes and is required to be a nonce by Scatter App, because the loomchain `nonce` is a number I have to create at least 6 bytes so I used the 256 hash algorithm to do that